### PR TITLE
improve site, start overhauling contributor guide

### PR DESCRIPTION
**Pull request by BenTheElder**
- normalize formatting for NOTEs
- overhaul contributing getting started guide
- improve ordering of resources page (bump down kubecon talks)
- cleanup outlines on various outlined things
- embrace `em` more
- start rewriting outdated project layout guide => detailed development guide

### CODE REVIEW

**Comment by BenTheElder** [comment]
https://deploy-preview-1955--k8s-kind.netlify.app/

File: site/content/docs/contributing/getting-started.md [hunk 1]
```diff
@@ -73,10 +67,22 @@ docker --version
 ```
 This documentation is written using Docker version 18.09.2.
 
+### Install Go (optional)
+
+KIND is written in [Go][golang], however our makefiles automatically ensure the
+correct version of go when building or testing.
+
+You may still wish to install go on your machine to make it easier to integrate
```
Comment (aojea) [inline site/content/docs/contributing/getting-started.md:78]: go here is inconsisten, you use Go in other places

Comment (BenTheElder) [reply site/content/docs/contributing/getting-started.md:78]: I think we should use it as `Go` when first talking about it so it's clear that it's a proper noun, but it's usually just written `go`.

File: site/content/docs/contributing/getting-started.md [hunk 2]
```diff
@@ -85,22 +91,27 @@ if there is any existing discussion or work related to your interests.
...
+
 Please reach out for bugs, feature requests, and other issues!  
 The maintainers of this project are reachable via:
 
-- [Kubernetes Slack] in the [#kind] channel
-- [filing an issue][file an issue]
-- The Kubernetes [SIG-Testing Mailing List]
+- [Kubernetes Slack] in the [#kind] channel (most active, along with the community)
+- The issue tracker by [filing an issue][file an issue]
+- The Kubernetes [SIG-Testing][SIG-Testing] [Mailing List][SIG-Testing Mailing List]
```
Comment (aojea) [inline site/content/docs/contributing/getting-started.md:107]: is this a copy paste issue or is the correct format? is not `[SIG-Testing][SIG-Testing Mailing List]`

File: site/content/docs/contributing/getting-started.md [hunk 3]
```diff
@@ -85,22 +91,27 @@ if there is any existing discussion or work related to your interests.
...
 
 Current maintainers are [@BenTheElder] and [@munnerz] - feel free to
```
Comment (aojea) [inline site/content/docs/contributing/getting-started.md:109]: should I? :smile:

Comment (BenTheElder) [reply site/content/docs/contributing/getting-started.md:109]: you can ping us :-)
or you want to be on the list? that's coming, we'll talk soon :-)

File: site/content/docs/contributing/getting-started.md (hunk 2 above)
Comment (BenTheElder) [reply site/content/docs/contributing/getting-started.md:107]: this is now two links.

Comment (BenTheElder) [reply site/content/docs/contributing/getting-started.md:107]: (check the rendered form in the preview)

File: site/content/docs/contributing/1.0-roadmap.md [hunk 4]
```diff
@@ -1,5 +1,5 @@
 ---
-title: "1.0 Roadmap"
+title: "1.0 Roadmap 🗺"
```
Comment (aojea) [inline site/content/docs/contributing/1.0-roadmap.md:2]: nice

File: site/content/docs/contributing/getting-started.md (hunk 3 above)
Comment (BenTheElder) [reply site/content/docs/contributing/getting-started.md:109]: as promised https://github.com/kubernetes-sigs/kind/pull/2025

**Pull request merged**